src/main/java/org/tron/common/overlay/discover/table/NodeEntry.java

```diff
@@ -70,7 +70,7 @@ public long getModified() {
   public boolean equals(Object o) {
     boolean ret = false;
 
-    if (o instanceof NodeEntry) {
+    if (this.getClass() == o.getClass()) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "o" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=853&issues=AWOg829GmXhJxGlVU5lL&open=AWOg829GmXhJxGlVU5lL)

[ISSUE_KEY]: AWOg829GmXhJxGlVU5lL